Sming/build.mk

```diff
@@ -148,14 +147,27 @@ else
 	CFLAGS		+= -Os -g
 endif
 
-CXXFLAGS = $(CFLAGS) -std=c++11 -felide-constructors
+CXXFLAGS = $(CFLAGS) -felide-constructors
 
 ifneq ($(STRICT),1)
 	CXXFLAGS += -Wno-reorder
 endif
 
 include $(ARCH_BASE)/build.mk
 
+# Detect compiler version
+DEBUG_VARS			+= GCC_VERSION
+GCC_VERSION			:= $(shell $(CC) -dumpversion)
+
+# Select C++17 if supported, defaulting to C++11 otherwise
+DEBUG_VARS			+= CPP_STD
+ifeq ($(GCC_VERSION),4.8.5)
+CPP_STD				:= c++11
+else
+CPP_STD				:= c++17
+endif
+CXXFLAGS			+= -std=$(CPP_STD)
```

Comment:
Note that I've elected to select C++17 for the current toolchains (except linux which doesn't support it). This has the potential to break some existing code, but AFAIK only insofar as it includes better checks and so not a bad thing IMHO. The alternative is to select C++17 for the new toolchain only. This could also be made user-selectable, but not sure that's necessarily a good move.

Reply:
.. although being able to select the standard would be useful for testing against C++20, etc.

Reply:
I've added support for this and included note in documentation.